src/DurationAwareTrait.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace RebelCode\Time;
+
+use Dhii\Util\String\StringableInterface as Stringable;
+use Exception as RootException;
+use InvalidArgumentException;
+
+/**
+ * Common functionality for objects that are aware of a duration.
+ *
+ * @since [*next-version*]
+ */
+trait DurationAwareTrait
+{
+    /**
+     * An integer number of seconds.
+     *
+     * @since [*next-version*]
+     *
+     * @var int
+     */
+    protected $duration;
+
+    /**
+     * Retrieves the timestamp.
+     *
+     * @since [*next-version*]
+     *
+     * @return int An integer number of seconds since unix epoch.
+     */
+    protected function _getDuration()
+    {
+        return $this->duration;
+    }
+
+    /**
+     * Sets the timestamp.
+     *
+     * @since [*next-version*]
+     *
+     * @param int|string|Stringable $duration The number of seconds since unix epoch. Negative numbers are allowed.
+     */
+    protected function _setDuration($duration)
+    {
+        try {
+            $this->duration = $this->_sanitizeTimestamp($duration);
```

Comment:
Why is `null` not allowed? I don't have a strong opinion about this yet, just asking.

Reply:
Good question. Does it make sense to "not have a duration"?

Reply:
Well, everywhere else it is possible to pass `null` - like this you can "unset" the whatever. Also, the initial duration is `null`, so I think not allowing an object to return to that same state can create weird consequences.